src/test/java/org/apache/commons/imaging/formats/tiff/TiffShortIntRoundTripTest.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging.formats.tiff;

+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.fail;

+import java.awt.image.BufferedImage;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.nio.file.Path;

+import org.apache.commons.imaging.FormatCompliance;
+import org.apache.commons.imaging.ImageReadException;
+import org.apache.commons.imaging.ImageWriteException;
+import org.apache.commons.imaging.common.ImageBuilder;
+import org.apache.commons.imaging.common.bytesource.ByteSourceFile;
+import org.apache.commons.imaging.formats.tiff.constants.TiffTagConstants;
+import org.apache.commons.imaging.formats.tiff.write.TiffImageWriterLossy;
+import org.apache.commons.imaging.formats.tiff.write.TiffOutputDirectory;
+import org.apache.commons.imaging.formats.tiff.write.TiffOutputSet;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;

+/**
+ * Performs a test in which a TIFF file with the special-purpose short-integer
+ * sample type is used to store data to a file. The file is then read to see if
+ * it matches the original values. The primary purpose of this test is to verify
+ * that the TIFF data reader classes behave correctly when reading raster data
+ * in various format.
+ */
+public class TiffShortIntRoundTripTest extends TiffBaseTest {

+    @TempDir
+    Path tempDir;

+    int width = 48;
+    int height = 23;

+    short[] sample = new short[width * height];

+    public TiffShortIntRoundTripTest() {
+        // populate the image data
+        for (int iCol = 0; iCol < width; iCol++) {
+            for (int iRow = 0; iRow < height; iRow++) {
+                final int index = iRow * width + iCol;
+                sample[index] = (short)index;
+            }
+        }

```

Comment:
I'm thinking about creating a JIRA ticket @gwlucastrig , to create a test-only component to generate images. Maybe a builder? Like...

```java
TestImagesBuilder
  .format(ImageFormats.TIFF)
  .withData((row, col, width, height) -> {
    return row * width + col;
  })
;
```

We could slowly add more methods to this test-class, to build custom metadata. Downside being that it's more code to maintain, and having a bug there could result in a bug in the code (but that's true and maybe even worse if we have to replicate code to create files in multiple test files...)

WDYT? Or maybe improve the existing `ImageBuilder`...

Reply:
If we were starting from scratch, this might be a good idea.  But given that we have a good (though not exhaustive) set of test files, I'm not sure it's worth our effort.

Also, TIFF has so many special cases that it might be unwieldy trying to fit them into a generic builder.  Also, at this point my main motivation for writing/reading special data formats is to improve Junit test case coverage...  so it's very specific and targeted to a particular need.

My plan for all my own future work is not to create any new test files for TIFF.  If we find some in a format that was not previously supported (as when somebody posts on on JIRA), it might make sense to add them to the collection...  I would want to do that because files created from outside Commons Imaging represent a "baseline" for comparing our own logic.

Reply:
Thanks @gwlucastrig! Will push that idea to the bottom of the backlog :+1: